test/unified_format.py

```diff
@@ -1880,7 +1872,15 @@ def _run_scenario(self, spec, uri=None):
         self.entity_map = EntityMapUtil(self)
         self.entity_map.create_entities_from_spec(self.TEST_SPEC.get("createEntities", []), uri=uri)
         # process initialData
-        self.insert_initial_data(self.TEST_SPEC.get("initialData", []))
+        if "initialData" in self.TEST_SPEC:
+            self.insert_initial_data(self.TEST_SPEC.get("initialData", []))
+            # advance cluster times of session entities,
+            # to ensure consistency in transactions against a sharded deployment,
+            cluster_time = self.client.admin.command("ping").get("$clusterTime")
+            if cluster_time:
+                for entity in self.entity_map._entities.values():
+                    if isinstance(entity, ClientSession):
+                        entity.advance_cluster_time(cluster_time)
```

Comment:
We need to do this for sessions created later on in `_testOperation_createEntities` too.

Reply:
Where is `_testOperation_createEntities` called from? Would we want to ping again before updating?

Reply:
We should only run ping once, store the cluster_time, then use it later when _testOperation_createEntities is called.